projection/core/src/main/scala/com/lightbend/lagom/internal/projection/ProjectionRegistry.scala

```diff
@@ -24,61 +30,72 @@ import scala.concurrent.duration._
 @ApiMayChange
 object ProjectionRegistry {
 
-  sealed trait WorkerStatus
-  case object Stopped extends WorkerStatus
-  case object Started extends WorkerStatus
-
-  sealed trait StateRequest
-  case object Stop  extends StateRequest
-  case object Start extends StateRequest
-
-  @ApiMayChange
-  case class ProjectionWorker(name: String, status: WorkerStatus)
-
-  @ApiMayChange
-  case class Projection(name: String, workers: Seq[ProjectionWorker])
-
+  case class StateRequestCommand(coordinates: WorkerCoordinates, requestedStatus: Status)
 }
 
 @ApiMayChange
 private[lagom] class ProjectionRegistry(system: ActorSystem) {
 
-  // A ProjectionRegistry is responsible for this node's ProjectionRegistryActor instance.
-  // TODO: decide what to do if/when the ProjectionRegistryActor dies (note the loss of references to projections).
   private val projectionRegistryRef: ActorRef = system.actorOf(ProjectionRegistryActor.props, "projection-registry")
   private lazy val clusterShardingSettings    = ClusterShardingSettings(system)
   private lazy val clusterDistribution        = ClusterDistribution(system)
 
   /**
    *
-   * @param streamName name of the stream this projection group consumes
-   * @param shardNames collection of partition names in the consumed stream
    * @param projectionName unique name identifying the projection group
+   * @param shardNames collection of partition names in the consumed stream
+   * @param projectionWorkerPropsFactory
    * @param runInRole
-   * @param projectionPropsFactory
    */
-  private[lagom] def registerProjectionGroup(
-      streamName: String,
-      shardNames: Set[String],
+  private[lagom] def registerProjection(
       projectionName: String,
-      runInRole: Option[String],
-      projectionPropsFactory: ActorRef => Props
+      shardNames: Set[String],
+      projectionWorkerPropsFactory: WorkerCoordinates => Props,
+      runInRole: Option[String] = None
   ): Unit = {
 
+    val projectionProps = (projectionRegistryActorRef: ActorRef) =>
+      WorkerHolderActor.props(projectionName, projectionWorkerPropsFactory, projectionRegistryActorRef)
+
     clusterShardingSettings.withRole(runInRole)
 
     clusterDistribution.start(
       projectionName,
-      projectionPropsFactory(projectionRegistryRef),
+      projectionProps(projectionRegistryRef),
```

Comment:
We don't need `projectionProps` function anymore. We can just pass

`WorkerHolderActor.props(projectionName, projectionWorkerPropsFactory, projectionRegistryRef)`

Reply:
We could inline but I like having the `val` indicating what that sorcery actually is. I would even improve the `val` declaration with the explicit type.

Reply:
Ok, but we don't need to have a function. That was my point.

We have a value, then we make a function to which we can pass it and in that function the only thing we do is to call the `props`. So, there is no reason to have such a function.